src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -450,112 +449,90 @@ protected virtual string GenerateUniqueId()
             }
         }
 
-        private void RenderUnboundAttribute(string attributeName, Chunk attributeValueChunk)
+        private void RenderUnboundAttribute(TagHelperAttributeTracker attribute)
         {
             // Render children to provide IntelliSense at design time. No need for the execution context logic, it's
             // a runtime feature.
             if (_designTimeMode)
             {
-                if (attributeValueChunk != null)
+                if (attribute.Value != null)
```

Comment:
Does the `attribute.Value == null` case still exist at design time (even though it does not at runtime)? Put another way,

1. Should the `_bodyVisitor.Accept()` call be unconditional?
2. Should the (missing) assertion that `attribute.Value` is non-`null` before or after the `_designTimeMode` check?